### PR TITLE
Add functions for deleting and updating templates in the database

### PR DESCRIPTION
**Pull request by paalka**
New dependencies: `labix.org/v2/mgo/bson` (should work if you have already installed mgo).

The error handling is not ideal atm, it does ex. not return the appropriate error code when someone attempts to update something, but does not provide any content to update.

### CODE REVIEW

File: db.go [hunk 1]
```diff
@@ -71,10 +72,26 @@ func getTemplatesByName(name string) []Template {
...
 	collection := sessionCopy.DB(DB_NAME).C(DB_COLL)
 	return collection.Insert(template)
 }
+
+func updateTemplate(template Template) error {
+	sessionCopy := mongoSession.Copy()
+	defer sessionCopy.Close()
+
+	collection := sessionCopy.DB(DB_NAME).C(DB_COLL)
+	return collection.Update(bson.M{"name": template.Name}, template)
```
Comment (sklirg) [inline db.go:88]: A name in a collection might not be unique.

File: db.go [hunk 2]
```diff
@@ -71,10 +72,26 @@ func getTemplatesByName(name string) []Template {
...
+}
+
+func deleteTemplate(template_name string) error {
+	sessionCopy := mongoSession.Copy()
+	defer sessionCopy.Close()
+
+	collection := sessionCopy.DB(DB_NAME).C(DB_COLL)
+	return collection.Remove(bson.M{"name": template_name})
```
Comment (sklirg) [inline db.go:96]: A name in a collection might not be unique.

File: handlers.go [hunk 3]
```diff
@@ -33,24 +33,47 @@ func GetAllTemplates(c *gin.Context) {
...
 func DeleteTemplate(c *gin.Context) {
+	template_name := c.Param("template_name")
+
+	err := deleteTemplate(template_name)
+	if err != nil {
+		error_msg := fmt.Sprintf("An error occured while attempting to delete: %v. ERROR: %v", template_name, err)
+		c.JSON(400, generateJSONErr(400, error_msg))
+		c.Abort()
+	} else {
+		c.JSON(200, "The template was successfully deleted.")
```
Comment (sklirg) [inline handlers.go]: Could return `204 No Content` instead, as the content is gone.

**Comment by sklirg** [comment]
Why'd ya add `publishing-templates`?

**Comment by sklirg** [comment]
Regarding uniqueness of names, we could add a simple check to stop insertion of preexisting names.

**Comment by paalka** [comment]
MongoDB seems to have the cabability of having unique indexes, so we can maybe use that instead.
InsertTemplate will anyways complain if an document with the same name exists already.

FWIW: The creation of an unique index can be done by using the command:
`db.templates.createIndex( { "name": 1 }, { unique: true })`

**Pull request merged**